src/ui/components/SignTransactionButton/SignTransactionButton.tsx

```diff
@@ -33,41 +41,57 @@ export const SignTransactionButton = React.forwardRef(
       buttonTitle,
       onClick,
       buttonKind = 'primary',
+      holdToSign,
       ...buttonProps
     }: React.ButtonHTMLAttributes<HTMLButtonElement> & {
       wallet: ExternallyOwnedAccount;
       buttonTitle?: React.ReactNode;
       buttonKind?: ButtonKind;
+      holdToSign: boolean;
     },
     ref: React.Ref<SendTxBtnHandle>
   ) {
+    const { preferences } = usePreferences();
     const hardwareSignRef = useRef<SignTransactionHandle | null>(null);
+    const { mutateAsync: sendTransactionInner, ...sendTxMutationInner } =
```

Comment:
What I tried here is to get the loading/completing/error state directly from different mutations. May be this can be refactored this way. Do we need to do it in this stage? I would probably do it separately, cause this will probably cause more potential bugs and we will need to test it carefully again before initial merge

Reply:
I also always try to avoid async functions inside onSucess callbacks if possible. Not sure is it a right approach.

Reply:
I think it's much more important to avoid mixing concerns. The UI elements of choice should not affect signing logic.

> cause this will probably cause more potential bugs

we should avoid merging something that "will cause more bugs" :)